cpp/CMakeLists.txt

```
@@ -321,37 +345,49 @@ endif(PER_THREAD_DEFAULT_STREAM)
 # - add gtest -------------------------------------------------------------------------------------
 
 if(BUILD_TESTS)
-    include(CTest)
-    include(ConfigureGoogleTest)
-
-    if(GTEST_FOUND)
-        message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
-        include_directories(${GTEST_INCLUDE_DIR})
-        add_subdirectory(${CMAKE_SOURCE_DIR}/tests)
-    else()
-        message(AUTHOR_WARNING "Google C++ Testing Framework (Google Test) not found: automated tests are disabled.")
-    endif(GTEST_FOUND)
+
+  CPMAddPackage(
+    NAME GTEST
+    GITHUB_REPOSITORY google/googletest
+    GIT_TAG release-1.10.0
+    VERSION 1.10.0
+    OPTIONS
+      "INSTALL_GTEST OFF"
+      "gtest_force_shared_crt"
+  )
+  set(GTEST_INCLUDE_DIR GTEST)
+  enable_testing()
+  
+  include_directories(${GTEST_INCLUDE_DIR})
+  add_subdirectory(${CMAKE_SOURCE_DIR}/tests)
```

I know this was already a problem, but `CMAKE_SOURCE_DIR` is unsafe if someone was to use `add_subdirectory(cudf)` since `CMAKE_SOURCE_DIR` points to the root project.
```suggestion
  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tests)
```

---

Even easier: ;)

```cmake
add_subdirectory(tests)
```